backend/de.metas.business/src/main/java/de/metas/bpartner/quick_input/service/BPartnerQuickInputService.java

```diff
@@ -488,83 +553,103 @@ private BPartnerComposite toBPartnerComposite(@NonNull final I_C_BPartner_QuickI
-				.orgId(OrgId.ofRepoId(template.getAD_Org_ID()))
-				.bpartner(bpartner)
-				.location(bpLocation)
-				.contacts(contacts)
-				.build();
+		return contacts;
+	}
+
+	private ArrayList<BPartnerLocation> getBPartnerLocations(final @NonNull I_C_BPartner_QuickInput template)
+	{
+		final BPartnerQuickInputId bpartnerQuickInputId = extractBpartnerQuickInputId(template);
+
+		final ArrayList<BPartnerLocation> bpartnerLocations = new ArrayList<>();
+
+		final List<I_C_BPartner_Location_QuickInput> bpartnerLocationTemplates = bpartnerQuickInputRepository.retrieveLocationsByQuickInputId(bpartnerQuickInputId);
+		for (final I_C_BPartner_Location_QuickInput bpartnerLocationTemplate : bpartnerLocationTemplates)
+		{
+			bpartnerLocations.add(BPartnerLocation.builder()
+										  .locationType(BPartnerLocationType.builder()
+																.billTo(bpartnerLocationTemplate.isBillTo())
+																.billToDefault(bpartnerLocationTemplate.isBillToDefault())
+																.shipTo(bpartnerLocationTemplate.isShipTo())
+																.shipToDefault(bpartnerLocationTemplate.isShipToDefault())
+																.build())
+										  .email(bpartnerLocationTemplate.getEMail())
+										  .fax(bpartnerLocationTemplate.getFax())
+										  .mobile(bpartnerLocationTemplate.getPhone2())
+										  .phone(bpartnerLocationTemplate.getPhone())
+										  .name(bpartnerLocationTemplate.getName())
+										  .gln(GLN.ofNullableString(bpartnerLocationTemplate.getGLN()))
+										  .existingLocationId(LocationId.ofRepoId(bpartnerLocationTemplate.getC_Location_ID()))
+										  .bpartnerName(bpartnerLocationTemplate.getBPartnerName())
+										  .setupPlaceNo(bpartnerLocationTemplate.getSetup_Place_No())
+										  .replicationLookupDefault(bpartnerLocationTemplate.isReplicationLookupDefault())
+										  .remitTo(bpartnerLocationTemplate.isRemitTo())
+										  .handOverLocation(bpartnerLocationTemplate.isHandOverLocation())
+										  .visitorsAddress(bpartnerLocationTemplate.isVisitorsAddress())
+										  .build());
+		}
+
+		return bpartnerLocations;
 	}
 
 	private void assertPriceListExists(
 			@NonNull final PricingSystemId pricingSystemId,
-			@NonNull final CountryId countryId,
+			@NonNull final Set<CountryId> countryIds,
 			@NonNull final SOTrx soTrx)
 	{
-		final PriceListId priceListId = priceListDAO.retrievePriceListIdByPricingSyst(pricingSystemId, countryId, soTrx);
-		if (priceListId == null)
+		for (final CountryId countryId : countryIds)
 		{
-			final String pricingSystemName = priceListDAO.getPricingSystemName(pricingSystemId);
-			throw new PriceListNotFoundException(pricingSystemName, soTrx);
+			final PriceListId priceListId = priceListDAO.retrievePriceListIdByPricingSyst(pricingSystemId, countryId, soTrx);
+
+			if (priceListId == null)
+			{
+				final String pricingSystemName = priceListDAO.getPricingSystemName(pricingSystemId);
+				throw new PriceListNotFoundException(pricingSystemName, soTrx);
```

Comment:
pls improve the error message: introduce the countryId**s** param. Make sure is displayed nicely.